tidy3d/components/grid/grid_spec.py

```diff
@@ -199,6 +199,47 @@ def _make_coords_initial(
         return center - size / 2 + np.arange(num_cells + 1) * dl_snapped
 
 
+class CustomGridBoundaries(GridSpec1d):
+    """Custom 1D grid supplied as a list of grid cell boundary coordinates. The grid must span
+    exactly the simulation domain size along the dimension, including PML layers, symmetric regions.
+
+    Example
+    -------
+    >>> grid_1d = CustomGridCoords(boundaries=[-0.2, 0.0, 0.2, 0.4, 0.5, 0.6, 0.7])
+    """
+
+    coords: Coords1D = pd.Field(
+        ...,
+        title="Grid boundary coordinates",
+        description="Grid boundariy coordinate which must span exactly the simulation domain size, "
+        "including PML layers. It is mostly used to duplicate grids to another simulation of the same size.",
+        units=MICROMETER,
+    )
+
+    def _make_coords_initial(
+        self,
+        axis: Axis,
+        structures: List[StructureType],
+        **kwargs,
+    ) -> Coords1D:
+        """Unused. Override ``make_coords`` directly as the
+        grid coordiate for the complete simulation domain is provided already.
+        """
```

Comment:
remove this? what's it for?

Reply:
just initializing a subclass?

Reply:
yeah, since it's an abstract method in the parent class.